### PR TITLE
Add AMDGPU v0.5 support

### PR DESCRIPTION
**Pull request by luraess**
- transition AMDGPU to HIP backend
- requires CellArray compat bump https://github.com/omlins/CellArrays.jl/pull/20

### CODE REVIEW

**Review by omlins — APPROVED** [review]
Looks good to me!

File: test/ParallelKernel/test_parallel.jl [hunk 1]
```diff
@@ -162,6 +162,7 @@ end
                         return
                     end
                     @parallel write_indices!(A);
+                    if $package == $PKG_AMDGPU AMDGPU.HIP.device_synchronize() end
```
Comment (luraess) [inline test/ParallelKernel/test_parallel.jl:165]: HOTFIX: device sync is somehow needed here and in later tests (mostly 3D). One should still investigate why further @omlins

Comment (luraess) [reply test/ParallelKernel/test_parallel.jl:165]: This relates to #106 description added in the issue.

Comment (omlins) [reply test/ParallelKernel/test_parallel.jl:165]: I will merge it as we need to release today. We can still check this...

Comment (luraess) [reply test/ParallelKernel/test_parallel.jl:165]: okay

**Pull request merged**